presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1652,6 +1652,33 @@ public boolean supportsMetadataDelete(ConnectorSession session, ConnectorTableHa
         return true;
     }
 
+    @Override
+    public Map<ColumnHandle, ColumnHandle> pushdownSubfieldPruning(
```

Comment:
Is this for prototyping or do we envision this being the standard for master?

Reply:
@tdcmeehan It depends on whether @highker's work on allowing connectors to participate in query planning lands soon enough. Ideally, we would present `project(filter(scan))` subplan to the connector and it will decide that it can implement all of it and return a new table scan node with table and column handles encoding projections and filters. If this can't happen soon enough, I do envision introducing a couple of new metadata APIs to allow for pushing down some information into the connector. Specifically,

- `pushdownSubfieldPruning` tells the connector that it can prune structs, maps and arrays
- `getTableLayouts(TupleDomain<ColumnHandleWithSubfieldPaths>)` allows the connector to implement simple filters on subfields, e.g. c[2] = 5.

Reply:
Sounds good.  I would add a comment discouraging its use before it reaches master.